client/components/wcc-settings-form/index.js

```diff
@@ -9,19 +9,76 @@ import { successNotice, errorNotice } from 'state/notices/actions';
 import * as SettingsActions from 'state/settings/actions';
 import * as PackagesActions from 'state/form/packages/actions';
 import { getFormErrors } from 'state/selectors';
+import { translate as __ } from 'lib/mixins/i18n';
+import FormButton from 'components/forms/form-button';
+import FormButtonsBar from 'components/forms/form-buttons-bar';
 
 const WCCSettingsForm = ( props ) => {
+	const currentStep = props.form.currentStep;
+	const currentStepLayout = props.layout[ currentStep ];
+
+	const renderCurrentStep = () => {
+		if ( ! currentStepLayout ) {
+			return <span>... spinner or something ...</span>;
+		}
+
+		return (
+			<WCCSettingsStep
+				{ ...props }
+				layout={ currentStepLayout }
+			/>
+		);
+	};
+
+	const renderActionButton = () => {
+		// TODO: extend <SaveForm> and use it in place of this
+		const label = ( currentStepLayout || {} ).action_label || __( 'Next' );
+		return (
+			<FormButtonsBar>
+				<FormButton type="button" onClick={ props.formActions.nextStep }>
+					{ label }
+				</FormButton>
+			</FormButtonsBar>
+		)
+	};
+
+	const renderTab = ( label, index ) => {
+		if ( index === currentStep ) {
+			return <b>{ label }</b>;
+		} else if ( index < currentStep ) {
+			return <a onClick={ () => props.formActions.goToStep( index ) }>{ label }</a>;
+		}
+		return <span>{ label }</span>;
+	};
+
+	const renderMultiStepForm = () => {
+		return (
+			<div>
+				<div>
+					<ul>
+						{ props.layout.map( ( step, index ) => (
+							<li>
+								{ renderTab( step.tab_title, index ) }
+							</li>
+						) ) }
+					</ul>
+				</div>
+				{ renderCurrentStep() }
+				{ renderActionButton() }
+			</div>
+		);
+	};
+
 	return (
 		<div>
 			<GlobalNotices id="notices" notices={ notices.list } />
-			{ props.layout.map( ( group, idx ) => (
-				<WCCSettingsGroup
+			{ 'step' === props.layout[ 0 ].type
+				? renderMultiStepForm()
+				: <WCCSettingsStep
```

Comment:
So, I generalized our current forms as just "a multi-step form of just one step". A bit hacky but it makes every use case we have fit into the same data model.